src/requests/cache/mod.rs

```diff
@@ -0,0 +1,15 @@
+use crate::cache_client::CacheClient;
+use crate::MomentoResult;
+
+pub mod set_add_elements;
+
+pub trait MomentoRequest {
+    type Response;
+}
+
+pub trait MomentoSendableRequest<R: MomentoRequest> {
```

Comment:
It's `pub` because it appears in the signature of the `public fn send_request` on the CacheClient. The trait was fine with it being private.

Reply:
Yeah, I was kind of saying this but pointing out that the trait's orientation is the root of the issue.

I think it's best to remove the duplicate trait because you can't remove the trait from the public signature. I didn't realize that yesterday - by making this function with a MomentoRequest trait public, you permit users to impl MomentoRequest for anything of their own, and for that implementation to be valid, send() would be required per the orientation of the api in this pr.

Since that mechanism is mandated, I'd put the fn back in MomentoRequest
```
/// An internal tool intended to be used by Momento request types. You can impl
/// this fn for your own types if you'd like to hand them to the Momento client
/// directly, but that is not currently an explicitly supported scenario and this trait
/// may change a little over time. If that's okay with you and you want to, impl away!
#[doc(hidden)]
fn send(
    self,
    cache_client: &CacheClient,
) -> impl std::future::Future<Output = MomentoResult<Self::Response>> + Send;
```

Reply:
I would be fine with this approach and I agree that due to the `pub` there's no point in having two traits.

I experimented with the `enum` path for a while and could not find a direction that seemed promising so I'm bailing on that unless you let me know that you want to discuss it further.